app/vmui/packages/vmui/src/components/CustomPanel/Views/TraceView.tsx

```diff
@@ -0,0 +1,59 @@
+import React, {FC, useState} from "preact/compat";
+import {TracingData} from "../../../api/types";
+import List from "@mui/material/List";
+import ListItem from "@mui/material/ListItem";
+import ListItemText from "@mui/material/ListItemText";
+import ListItemIcon from "@mui/material/ListItemIcon";
+import ListItemButton from "@mui/material/ListItemButton";
+import Collapse from "@mui/material/Collapse";
+import Box from "@mui/material/Box";
+import ExpandLess from "@mui/icons-material/ExpandLess";
+import AddCircleRoundedIcon from "@mui/icons-material/AddCircleRounded";
+import {recursiveComponent} from "../NestedNav/NestedNav";
+
+
+interface TraceViewProps {
+  tracingData: TracingData;
+}
+
+interface OpenLevels {
+  [x: number]: boolean
+}
+
+const TraceView: FC<TraceViewProps> = ({tracingData}) => {
+
+  const [openLevels, setOpenLevels] = useState({} as OpenLevels);
+  const handleClick = (param: number) => {
+    setOpenLevels((prevState:OpenLevels) => ({...prevState, [param]: !prevState[param]}));
+  };
+
+  return (<List sx={{ width: "100%" }} component="nav">
+    <Box sx={{ bgcolor: "rgba(201, 227, 246, 0.4)" }}>
+      <ListItem onClick={() => handleClick(tracingData.duration_msec)} sx={{p: 0}}>
```

Comment:
I cannot get the idea why the top level of the trace needs a separate rendering logic comparing to lower levels of the trace. Probably it would be better to pass the complete `trace` object from the response to `recursiveComponent`, so it draws generic tree view?

Reply:
simplified NestedNav component, at this moment it is one simple component